.github/workflows/release-build.yml

```diff
@@ -0,0 +1,96 @@
+name: Build Release Artifacts
+
+on:
+  release:
+    types:
+      - published
+  workflow_dispatch:
+
+jobs:
+  build-release-artifacts:
+    name: "Build Release Artifacts"
+    runs-on: ${{ matrix.operating-system }}
+    strategy:
+      fail-fast: false
+      matrix:
+        php-version:
+          - "8.1"
+        operating-system:
+          - "ubuntu-latest"
+          - "macos-latest"
+    steps:
+      - name: "Checkout"
+        uses: "actions/checkout@v4"
+
+      - name: "Install PHP"
+        uses: "shivammathur/setup-php@v2"
+        with:
+          coverage: none
+          tools: composer:v2
+          php-version: "${{ matrix.php-version }}"
+          ini-values: memory_limit=-1
+
+      - name: "Get Composer Cache Directory"
+        id: composer-cache
+        run: |
+          echo "dir=$(composer config cache-files-dir)" >> $GITHUB_OUTPUT
+
+      - name: "Cache Composer dependencies"
+        uses: "actions/cache@v3"
+        with:
+          path: "${{ steps.composer-cache.outputs.dir }}"
+          key: "php-${{ matrix.php-version }}-locked-composer-${{ hashFiles('**/composer.lock') }}"
+          restore-keys: |
+            php-${{ matrix.php-version }}-locked-composer-
+      - name: "Install locked dependencies"
+        run: "composer install --no-interaction --no-progress"
+
+      # Cache downloaded source
+      - id: cache-download
+        uses: actions/cache@v3
+        with:
+          path: downloads
+          key: php-${{ matrix.php-version }}-dependencies
+
+      # If there's no dependencies cache, fetch sources
+      - if: steps.cache-download.outputs.cache-hit != 'true'
```

Comment:
What if dependencies have been updated (e.g.: security fix) since the last retrieval?
Shouldn't the command always be run?

Reply:
It depends on what our definition of a security fix is. If there are security updates for dependent libraries included in the compilation results, this is generally not necessary because this Action only distributes spc binaries. If it is a security update for the project itself, it has nothing to do with the cache of the download itself.

Reply:
I think we could use a lock hash similar to the composer one with the usage of download lock: `DOWNLOAD_PATH . '/.lock.json'`

Reply:
`.lock.json` is now more like a download meta of downloaded files. I am not very clear about the reason and method of using it as a cache key.

Reply:
The reason I can imagine is a stale cache with security issues, but we can add a clean-up workflow to be scheduled ie once a week, that would cover this case too

Reply:
If we just make a weekly build action, I think it could be done without any download caches. What I want to know more is, can the Action Cache take effect in the Fork warehouse?

Reply:
AFAIK cache & artifacts are not shared with the forks, the only way it is possible is when the fork does PR to the main repository and must use the same workflow, cause cache is per workflow file and jobs, but it can be shared within the same workflow (requires marking jobs are depending on each other).

Reply:
I think that working on the https://github.com/crazywhalecc/static-php-cli/issues/187 idea would be the only way to solve this problem on the DX side, in this repository we could setup weekly builds to ensure security fixes are always included in binaries.